include/device.h

```diff
@@ -250,38 +240,28 @@ struct device_pm {
 };
 
 /**
- * @brief Static device information (In ROM) Per driver instance
+ * @brief Runtime device structure (in memory) per driver instance
  *
  * @param name name of the device
  * @param init init function for the driver
  * @param config_info address of driver instance config information
- */
-struct device_config {
-	const char *name;
-	int (*init)(struct device *device);
-#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
-	int (*device_pm_control)(struct device *device, u32_t command,
-				 void *context, device_pm_cb cb, void *arg);
-	struct device_pm *pm;
-#endif
-	const void *config_info;
-};
-
-/**
- * @brief Runtime device structure (In memory) Per driver instance
  * @param device_config Build time config information
  * @param driver_api pointer to structure containing the API functions for
- * the device type. This pointer is filled in by the driver at init time.
+ * the device type.
  * @param driver_data driver instance data. For driver use only
  */
 struct device {
-	const struct device_config *config;
+	const char *name;
+	const void *config_info;
 	const void *driver_api;
-	void *driver_data;
+	void * const driver_data;
```

Comment:
per #22941, what's the plan for the device dependency here? I mean the APIs you want to expose to other sub-systems, AFAIK Linux will put `struct device *parent` here to represent the dependency, but the dependency building process belongs to DTS. I'm considering the APIs of PM about how to utilize these dependencies to power management system, it will be good to know the APIs of device dependency.

Reply:
It is not part of this RFC. See #22941 for the planned changes.